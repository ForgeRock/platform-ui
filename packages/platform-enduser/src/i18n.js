/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
