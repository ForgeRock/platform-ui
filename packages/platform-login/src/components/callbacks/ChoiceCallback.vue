<!-- Copyright 2019 ForgeRock AS. All Rights Reserved

Use of this code requires a commercial software license with ForgeRock AS.
or with one of its affiliates. All use shall be exclusively subject
to such license between the licensee and ForgeRock AS. -->
<template>
  <div>
    <label>{{ callback.getPrompt() }}</label>
    <BFormSelect
      :options="options"
      :name="name"
      v-model="selected"
      class="mb-2"
      @change="setValue" />
  </div>
</template>

<script>
import { map } from 'lodash';
import { BFormSelect } from 'bootstrap-vue';

export default {
  components: {
    BFormSelect,
  },
  props: {
    callback: {
      type: Object,
      required: true,
    },
    index: {
      type: Number,
      default: 0,
    },
  },
  mounted() {
    this.selected = this.callback.getDefaultChoice();

    this.name = `callback_${this.index}`;

    this.loadOptions();
  },
  data() {
    return {
      options: [],
      name: '',
      selected: null,
      choices: this.callback.getChoices(),
    };
  },
  methods: {
    loadOptions() {
      this.name = `callback_${this.index}`;
      this.options = map(this.choices, (item, itemIndex) => ({
        text: item,
        value: itemIndex,
      }));
    },
    setValue() {
      this.callback.setInputValue(this.selected);
    },
  },
};
</script>
