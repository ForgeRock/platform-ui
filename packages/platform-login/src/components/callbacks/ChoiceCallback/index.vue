<!-- Copyright 2019-2020 ForgeRock AS. All Rights Reserved
Use of this code requires a commercial software license with ForgeRock AS.
or with one of its affiliates. All use shall be exclusively subject
to such license between the licensee and ForgeRock AS. -->
<template>
  <div>
    <FrField :field="selected" />
  </div>
</template>
<script>
import { map } from 'lodash';
import FrField from '@forgerock/platform-shared/src/components/Field';

export default {
  components: {
    FrField,
  },
  props: {
    callback: {
      type: Object,
      required: true,
    },
    index: {
      type: Number,
      default: 0,
    },
  },
  mounted() {
    this.loadOptions();
  },
  data() {
    return {
      selected: {
        key: `callback_${this.index}`,
        title: this.callback.getPrompt(),
        type: 'select',
        value: this.callback.getDefaultChoice(),
        options: [],
      },
      choices: this.callback.getChoices(),
    };
  },
  watch: {
    selected(newVal) {
      this.callback.setInputValue(newVal);
    },
  },
  methods: {
    loadOptions() {
      this.selected.options = map(this.choices, (item, itemIndex) => ({
        text: item,
        value: itemIndex.toString(),
      }));
    },
  },
};
</script>
