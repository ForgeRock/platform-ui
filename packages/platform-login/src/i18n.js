/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
