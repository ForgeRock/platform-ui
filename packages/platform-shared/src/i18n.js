/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
