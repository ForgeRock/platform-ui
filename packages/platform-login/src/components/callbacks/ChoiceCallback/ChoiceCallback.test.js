/**
 * @license
 * Copyright (c) 2020 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { mount } from '@vue/test-utils';
import ChoiceCallback from '@/components/callbacks/ChoiceCallback';
import i18n from '@/i18n';

describe('ChoiceCallback', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(ChoiceCallback, {
      i18n,
      propsData: {
        callback: {
          getChoices: () => ['a', 'b', 'c', 'd'],
          getDefaultChoice: () => 0,
          getPrompt: () => 'select',
          setInputValue: jest.fn(),
        },
        index: 5,
      },
    });
  });

  it('Load ChoiceCallback component', () => {
    expect(wrapper.name()).toEqual('ChoiceCallback');
  });

  it('Sets selected and choices/options data', () => {
    expect(wrapper.vm.$data.selected).toMatchObject(
      {
        key: 'callback_5',
        title: 'select',
        type: 'select',
        value: 0,
        options: [
          { text: 'a', value: 0 },
          { text: 'b', value: 1 },
          { text: 'c', value: 2 },
          { text: 'd', value: 3 }],
      },
    );
  });

  it('Sets prompt on DOM', () => {
    const select = wrapper.find('.multiselect');
    const elements = () => select.findAll('.multiselect__option');

    select.trigger('click');
    elements().at(2).trigger('click');

    expect(wrapper.vm.$data.selected).toMatchObject({ value: 2 });
    expect(wrapper.vm.$props.callback.setInputValue).toHaveBeenLastCalledWith(2);
  });
});
