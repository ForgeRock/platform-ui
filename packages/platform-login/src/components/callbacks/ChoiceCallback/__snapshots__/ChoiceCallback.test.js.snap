// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceCallback Sets selected and choices/options data: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <FrField",
    "key=1",
    "modelValue=0",
    "onUpdate:modelValue=fn",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (2)",
    "
",
    " at <SelectInput",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (3)",
    "
",
    " at <VueMultiselect",
    "id="floatingLabelInput5"",
    "ref="vms"",
    "value=""",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput5"",
    "name="callback_5"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <SelectInput",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "appear" with v-bind value \`false\` will render appear="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "id="floatingLabelInput5"",
    "ref="vms"",
    "value=""",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput5"",
    "name="callback_5"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <SelectInput",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "persisted" with v-bind value \`false\` will render persisted="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "id="floatingLabelInput5"",
    "ref="vms"",
    "value=""",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput5"",
    "name="callback_5"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <SelectInput",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_ENUMERATED_COERCION) Enumerated attribute "spellcheck" with v-bind value \`false\` will render the value as-is instead of coercing the value to "true" in Vue 3. Always use explicit "true" or "false" values for enumerated attributes. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_ENUMERATED_COERCION: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <VueMultiselect",
    "id="floatingLabelInput5"",
    "ref="vms"",
    "value=""",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput5"",
    "name="callback_5"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <SelectInput",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "autofocus" with v-bind value \`false\` will render autofocus="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <FrField",
    "key=1",
    "type="select"",
    "autofocus=false",
    " ...",
    ">",
    "
",
    " at <ChoiceCallback",
    "callback=",
    {
      "getChoices": [Function],
      "getDefaultChoice": [Function],
      "getPrompt": [Function],
      "setInputValue": [MockFunction],
    },
    "index=5",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
